### DISCUSSION TITLE
this is fastapi running error

### DISCUSSION THREAD

**Discussion by songzh1100** [fastapi/fastapi#12958] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI,APIRouter
import uvicorn

from demo import demo_router

app = FastAPI()

demo_router=APIRouter()

@demo_router.get('/test1')
async def demo2():
    """测试"""
    return 'hello'

app.include_router(demo_router)

if __name__ == '__main__':
    uvicorn.run(app='domain:app', host='0.0.0.0', port=8003, workers=1)
```

### Description

[33mWARNING[0m:  Invalid HTTP request received.
[33mWARNING[0m:  Invalid HTTP request received.
[33mWARNING[0m:  Invalid HTTP request received.
Accept failed on a socket
socket: <asyncio.TransportSocket fd=148, family=AddressFamily.AF_INET, type=Sock
etKind.SOCK_STREAM, proto=6, laddr=('0.0.0.0', 5509)>
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\envs\fstyxt\lib\asyncio\proactor_events.py", li
ne 801, in loop
    conn, addr = f.result()
  File "C:\ProgramData\Anaconda3\envs\fstyxt\lib\asyncio\windows_events.py", lin
e 808, in _poll
    value = callback(transferred, key, ov)
  File "C:\ProgramData\Anaconda3\envs\fstyxt\lib\asyncio\windows_events.py", lin
e 551, in finish_accept
    ov.getresult()
OSError: [WinError 64] 指定的网络名不再可用。
Task exception was never retrieved
future: <Task finished name='Task-42' coro=<IocpProactor.accept.<locals>.accept_
coro() done, defined at C:\ProgramData\Anaconda3\envs\fstyxt\lib\asyncio\windows
_events.py:559> exception=OSError(22, '指定的网络名不再可用。', None, 64, None)>

Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\envs\fstyxt\lib\asyncio\windows_events.py", lin
e 562, in accept_coro
    await future
  File "C:\ProgramData\Anaconda3\envs\fstyxt\lib\asyncio\proactor_events.py", li
ne 801, in loop
    conn, addr = f.result()
  File "C:\ProgramData\Anaconda3\envs\fstyxt\lib\asyncio\windows_events.py", lin
e 808, in _poll
    value = callback(transferred, key, ov)
  File "C:\ProgramData\Anaconda3\envs\fstyxt\lib\asyncio\windows_events.py", lin
e 551, in finish_accept
    ov.getresult()
OSError: [WinError 64] 指定的网络名不再可用。

### Operating System

Windows

### Operating System Details

windows server 2008 r2 sp1

### FastAPI Version

0.115.5

### Pydantic Version

1.10.5

### Python Version

3.8.5

### Additional Context

What causes the program to report "The specified network name is no longer available" after running for a period of time
<img width="389" alt="20241122084117" src="https://github.com/user-attachments/assets/67a668c2-9311-4bc1-ac72-a5521171e8db">

**Comment by estebanx64** [comment]
Hi @songzh1100 👋

Unfortunately, the image you uploaded before isn't loading, can you upload it again to give us more context? 🤓

**Reply by songzh1100** [reply]
Thank you for your answer. The picture has been re-uploaded. This problem has been troubling us for a long time, and we don't know what went wrong.

**Reply by estebanx64** [reply]
@songzh1100 

How are you running the fastapi application? I'm seeing you are using an Anaconda environment directly in your windows machine, is it right?

I have a couple of questions. 🤓
Do you have any service or code that is special or the reason for this failure and can you share it here?
Are you running fastapi in a container or just as a process on your windows machine?
Are you running fastapi with uvicorn, gunicorn, and with a proxy as a load balancer like nginx, traaefik?
I'm not a windows expert, but have you checked for firewall rules or any restriction or limit by current network stats?
Is this a production-grade app? how many clients are sending requests?

A detailed explanation of your setup would be good 🙇

**Reply by estebanx64** [reply, marked as answer]
Hi @songzh1100. 👋

After doing some investigation, it seems this issue isn't related to fastapi itself, but to a misconfiguration in Windows or your setup for the deployment of the fastapi for a high concurrency request. 🤓

Some references I found.
 
 https://stackoverflow.com/questions/77049215/oserror-winerror-64-the-specified-network-name-is-no-longer-available-fastap
 https://github.com/RasaHQ/rasa/issues/10546

I have some recommendations, Make sure to have the ports/firewall/windows/network rules configured to allow the communication in your app.
I would recommend you read the documentation about deployment for the production environment if you have a significant concurrency of requests.
https://fastapi.tiangolo.com/deployment/

There is a uvicorn section for deployment too.
https://www.uvicorn.org/deployment/
There are some values like number of workers, limit-concurrency, etc.

And finally, you'll need a reverse proxy/load balance like nginx to handle the concurrency as well.

I hope these recommendations can help you to fix your issues! 🙇
